### PR TITLE
Bug/jenkins 38023 show expandable path

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- See [JENKINS-38023](https://issues.jenkins-ci.org/browse/JENKINS-38023).
- JDL PR: jenkinsci/jenkins-design-language#105
- TODO: 
   - [x] Sort out a few tests that need updating given that there's a new component in play
   - [x] Fix an issue where the favorite card displays the wrong "displayName" for a favorited branch (it's the branch name, not the parent MBP's displayName). Working with @vivek on this.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
regarding folders functionality - real nice. If you sort out other issues and ATH etc - I think :bee: form me (provisional). Very cool.

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx [hunk 1]
```diff
@@ -57,11 +51,11 @@ class RunDetailsHeader extends Component {
...
                 <h4>
                     <a onClick={() => this.handleOrganizationClick()}>{run.organization}</a>
-                    &nbsp;/&nbsp;
-                    { cleanFullName && `${cleanFullName} / `}
-                    <a onClick={() => this.handleNameClick()}>{displayName}</a>
-                    &nbsp;
-                    #{run.id}
+                    <span>&nbsp;/&nbsp;</span>
+                    <a className="path-link" onClick={() => this.handleNameClick()}>
+                        <ExpandablePath path={fullDisplayName} iconSize={20} hideFirst />
```
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: nice

File: blueocean-personalization/src/test/js/components/DashboardCards-spec.jsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ describe('DashboardCards', () => {
         testUtils.unbindAll();
     });
 
-    it('renders without error for empty props', () => {
+    xit('renders without error for empty props', () => {
```
Comment (michaelneale) [inline blueocean-personalization/src/test/js/components/DashboardCards-spec.jsx]: deliberate?

**Comment by cliffmeyers** [comment]
@vivek changes generally look good but I have two questions:

- Just confirming with you that it's expected that each element in `fullDisplayName` is URI-encoded? If I have a folder name such as "Simple Pipeline Jobs" the spaces are encoded as `%20`; this is fine, I will account for it in the UI code.
- I noticed that in the `/favorites` API that the an item with `_class=BranchImpl` has a `fullDisplayName` where the branch name is not URI-encoded. For example a multibranch project with a branch named according to git-flow style that was favorited looks like this:

```
"displayName": "experiment/build-locally-docker",
"fullDisplayName": "jdl1/experiment/build-locally-docker",
"fullName": "jdl1/experiment%2Fbuild-locally-docker",
```

I think perhaps that the `fullDisplayName` should actually be "jdl1/experiment%2Fbuild-locally-docker" in this case?

**Comment by michaelneale** [comment]
@cliffmeyers ready to review again? I see some changes by @vivek. Slightly worried about encoding rearing its ugly head again, need to treat very very carefully here.

**Comment by cliffmeyers** [comment]
@michaelneale I would say let's wait for a potential change from @vivek first, as that should definitely get this one entirely wrapped up.

**Comment by vivek** [comment]
@cliffmeyers that would be a bug. I will get a fix in shortly. @michaelneale yeah no easy way out, we pay the price of allowing path delimiter in the display name.

**Comment by vivek** [comment]
@cliffmeyers I think I fixed it. Please check it out.

**Comment by cliffmeyers** [comment]
@vivek pulled this down and tested it: looks good to me. @michaelneale perhaps you'd like to run through this one now? I think it's good to go.

**Comment by cliffmeyers** [comment]
ATH passed locally.
```
2016-10-20 15:46:53,771 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - >> Test suite "done". Okay to start next test.
2016-10-20 15:46:53,846 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-10-20 15:46:53,847 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - OK. 502  total assertions passed. (4m 30s)
2016-10-20 15:46:53,888 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [15:46:53] Finished 'default' after 4.5 min
```

**Comment by michaelneale** [comment]
functionally seems good: https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/201/ is ATH running for this so provisional :bee:

**Pull request merged**